executing/_position_node_finder.py

```diff
@@ -213,6 +215,31 @@ def test_for_decorator(self, node: EnhancedAST, index: int) -> None:
                 if sys.version_info < (3, 12):
                     index += 4
 
+    def fix_result(
+        self, node: EnhancedAST, instruction: dis.Instruction
+    ) -> EnhancedAST:
+        if (
+            sys.version_info >= (3, 12, 5)
+            and instruction.opname in ("GET_ITER", "FOR_ITER")
+            and isinstance(node.parent, ast.For)
```

Comment:
```suggestion
            and isinstance(node.parent, ast.For)
            and node is node.parent.iter
```

What about comprehensions?

Reply:
They node positions for comprehensions are not affected.

```python
class Foo:
    def __iter__(self):
        assert False

a = [x for x in Foo()]
```

output (Python 3.12.5):
```python
Traceback (most recent call last):
  File "/home/frank/projects/executing/codi.py", line 7, in <module>
    a = [x for x in Foo()]
        ^^^^^^^^^^^^^^^^^^
  File "/home/frank/projects/executing/codi.py", line 5, in __iter__
    assert False
           ^^^^^
AssertionError
```

Reply:
Based on the apparently desired behaviour for `For`, that seems like a Python bug lol

Reply:
I don't know, `For` is usually a multiline statement and comprehensions are usually on a single line. I will ask :smile: